### DISCUSSION TITLE
Defining an orthotropic material law

### DISCUSSION THREAD

**Discussion by leopeuziat** [thelfer/tfel#334] [Q&A] [answered]
Hi,
I would like to define an orthotropic material law. I didn't find examples in the documentation.
Thanks by advance,
Leo

**Comment by thelfer** [comment, marked as answer]
Hi @leopeuziat,

You have a least four examples in the gallery:

- https://thelfer.github.io/tfel/web/orthotropiclinearhardeningplasticity.html
- https://thelfer.github.io/tfel/web/barlat-yld2004.html
- https://thelfer.github.io/tfel/web/MericCailletaudSingleCrystalPlasticity.html
- https://thelfer.github.io/tfel/web/ExplicitBerveillerZaouiPolyCrystals.html

You can also find more information on those pages:

- https://thelfer.github.io/tfel/web/tensors.html
- https://thelfer.github.io/tfel/web/tfel-material.html

Last but not least, you may want to have a look at the help of the following keywords:

- `mfront --help-keyword=Implicit:@OrthotropicBehaviour`
- `mfront --help-keyword=Implicit:@ComputeStiffnessTensor`
- `mfront --help-keyword=Implicit:@ComputeThermalExpansion`
- `mfront --help-keyword=Implicit:@HillTensor`

(you may want to replace Implicit by the name of the DSL you plan to use).

Finally, the documentation of the [StandardElastoViscoPlasticity brick](https://thelfer.github.io/tfel/web/StandardElastoViscoPlasticityBrick.html) describes various orthotropic stress criteria and kinematic hardening rules which are ready to use.

Althougth not a reliable source, you can also have a look at the tests in the source. Apparently, 93 tests are related to orthotropic behaviours:

```bash
$ grep -l Orthotropic $(find . -name "*.mfront") | wc -l
```

**Reply by leopeuziat** [reply]
I was speaking about @Parser MaterialLaw; for instance orthotropic thermal expansion coefficient

**Reply by thelfer** [reply]
Ok, my bad. There isn't such example.
Orthotropy is not handled by material properties. For the mean thermal expansion coefficients, you have to create three files, one for each material direction.
Later, in a behaviour, you can use the `@ComputeThermalExpansion` as follows:

```cxx
@ComputeThermalExpansion {"ThermalExpansion1.mfront", "ThermalExpansion2.mfront", "ThermalExpansion3.mfront"};
```
